src/assets/templates/typescript/src/steps/user-field-equals.ts

```diff
@@ -60,25 +70,35 @@ export class UserFieldEqualsStep extends BaseStep implements StepInterface {
       return this.error('There was a problem connecting to JSON Placeholder.');
     }
 
-    if (apiRes.body.length === 0) {
-      // If no results were found, return an error.
-      return this.error('No user found for email %s', [email]);
-    } else if (!apiRes.body[0].hasOwnProperty(field)) {
-      // If the given field does not exist on the user, return an error.
-      return this.error('The %s field does not exist on user %s', [field, email]);
-    } else if (apiRes.body[0][field] == expectedValue) {
-      // If the value of the field matches expectations, pass.
-      return this.pass('The %s field was set to %s, as expected', [
-        field,
-        apiRes.body[0][field],
-      ]);
-    } else {
-      // If the value of the field does not match expectations, fail.
-      return this.fail('Expected %s field to be %s, but it was actually %s', [
-        field,
-        expectedValue,
-        apiRes.body[0][field],
-      ]);
+    try {
+      if (apiRes.body.length === 0) {
+        // If no results were found, return an error.
+        return this.error('No user found for email %s', [email]);
+      } else if (!apiRes.body[0].hasOwnProperty(field)) {
+        // If the given field does not exist on the user, return an error.
+        return this.error('The %s field does not exist on user %s', [field, email]);
+      } else if (this.compare(operator, apiRes.body[0][field], expectedValue)) {
+        // If the value of the field matches expectations, pass.
+        return this.pass(this.operatorSuccessMessages[operator.replace(/\s/g, '').toLowerCase()], [
```

Comment:
- [ ] Believe the `.replace()` and `.toLowerCase()` calls may no longer be required, as the keys now reflect the operator values.